### PR TITLE
[DPE-1777] Feature/cos support

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

PG-k8s missing COS support as per 

## Solution

Added COS support to PG-k8s

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -41,6 +41,10 @@ provides:
     interface: pgsql
   db-admin:
     interface: pgsql
+  metrics-endpoint:
+    interface: prometheus_scrape
+  grafana-dashboard:
+    interface: grafana_dashboard
```
Comment (dragomirp) [inline metadata.yaml:47]: Should these be without limits?

**Review by dragomirp — APPROVED** [review]

File: src/constants.py [hunk 2]
```diff
@@ -10,12 +10,16 @@
 REPLICATION_PASSWORD_KEY = "replication-password"
 REWIND_USER = "rewind"
 REWIND_PASSWORD_KEY = "rewind-password"
+MONITORING_USER = "monitoring"
+MONITORING_PASSWORD_KEY = "monitoring-password"
```
Comment (taurus-forever) [inline src/constants.py:14]: Q: should MONITORING_USER be in the list of SYSTEM_USERS?

File: tests/unit/test_charm.py [hunk 3]
```diff
@@ -524,6 +525,22 @@ def test_postgresql_layer(self, _, __, ___):
...
+                    "command": "/start-exporter.sh",
+                    "startup": "enabled",
+                    "after": [self._postgresql_service],
+                    "user": "postgres",
+                    "group": "postgres",
+                    "environment": {
+                        "DATA_SOURCE_NAME": (
+                            f"user=monitoring "
+                            f"password={self.charm.get_secret('app', 'monitoring-password')} "
+                            "host=/var/run/postgresql port=5432 database=postgres"
```
Comment (taurus-forever) [inline tests/unit/test_charm.py:540]: Should you use constant DATABASE_PORT instead of hardcoded port?

File: src/charm.py [hunk 4]
```diff
@@ -902,6 +933,22 @@ def _postgresql_layer(self) -> Layer:
...
+                    "command": "/start-exporter.sh",
+                    "startup": "enabled",
+                    "after": [self._postgresql_service],
+                    "user": WORKLOAD_OS_USER,
+                    "group": WORKLOAD_OS_GROUP,
+                    "environment": {
+                        "DATA_SOURCE_NAME": (
+                            f"user={MONITORING_USER} "
+                            f"password={self.get_secret('app', MONITORING_PASSWORD_KEY)} "
+                            "host=/var/run/postgresql port=5432 database=postgres"
```
Comment (taurus-forever) [inline src/charm.py:948]: Should you use constant DATABASE_PORT instead of hardcoded port?

**Review by taurus-forever — COMMENTED** [review]
LGTM, I will keep the latest vote here to Marcelo.
P.S. see some minor nitpicking.

File: tests/unit/test_charm.py (hunk 3 above)
Comment (paulomach) [reply tests/unit/test_charm.py:540]: the exporter is using a socket connection, but apparently validates if a port is defined in the envvar, independently of the value

File: src/constants.py (hunk 2 above)
Comment (paulomach) [reply src/constants.py:14]: @marceloneppel

Comment (paulomach) [reply src/constants.py:14]: 1a182bc

**Pull request merged**